src/main/java/teammates/ui/controller/InstructorStudentRecordsPageData.java

```diff
@@ -1,24 +1,99 @@
 package teammates.ui.controller;
 
+import java.util.ArrayList;
 import java.util.List;
 
 import teammates.common.datatransfer.AccountAttributes;
 import teammates.common.datatransfer.CommentAttributes;
 import teammates.common.datatransfer.InstructorAttributes;
-import teammates.common.datatransfer.SessionAttributes;
+import teammates.common.datatransfer.StudentAttributes;
 import teammates.common.datatransfer.StudentProfileAttributes;
+import teammates.common.util.Const;
+import teammates.common.util.Sanitizer;
+import teammates.ui.template.Comment;
+import teammates.ui.template.CommentsForStudentsTable;
+import teammates.ui.template.StudentProfile;
 
 public class InstructorStudentRecordsPageData extends PageData {
 
-    public String courseId;
-    public InstructorAttributes currentInstructor;
-    public StudentProfileAttributes studentProfile;
-    public List<CommentAttributes> comments;
-    public List<SessionAttributes> sessions;
-    public String showCommentBox;
+    private String courseId;
+    private String studentName;
+    private String studentEmail;
+    private String googleId;
+    private String showCommentBox;
+    private StudentProfile studentProfile;
+    public StudentProfileAttributes spa; // used for testing admin message
+    private List<CommentsForStudentsTable> commentsForStudentTable;
+    private List<String> sessionNames;
 
-    public InstructorStudentRecordsPageData(AccountAttributes account) {
-        super(account);
+    public InstructorStudentRecordsPageData(AccountAttributes account, StudentAttributes student,
+                                            String courseId, String showCommentBox, StudentProfileAttributes spa,
+                                            List<CommentAttributes> comments, List<String> sessionNames,
+                                            InstructorAttributes instructor) {
+        super(account, student);
+        this.courseId = courseId;
+        this.studentName = Sanitizer.sanitizeForHtml(student.name);
+        this.studentEmail = student.email;
+        this.googleId = account.googleId;
+        this.showCommentBox = showCommentBox;
+        if (spa != null) {
+            this.spa = spa;
+            String pictureUrl = Const.ActionURIs.STUDENT_PROFILE_PICTURE
+                              + "?" + Const.ParamsNames.BLOB_KEY + "=" + spa.pictureKey
+                              + "&" + Const.ParamsNames.USER_ID + "=" + account.googleId;
+            if (spa.pictureKey.isEmpty()) {
+                pictureUrl = Const.SystemParams.DEFAULT_PROFILE_PICTURE_PATH;
+            }
+            this.studentProfile = new StudentProfile(this.studentName, spa, pictureUrl);
```

Comment:
Just checking, but is it necessary to use the sanitized version of `studentName` here?